internal/cri/opts/spec_linux_opts.go

```diff
@@ -68,12 +68,20 @@ func WithMounts(osi osinterface.OS, config *runtime.ContainerConfig, extra []*ru
 		sort.Stable(orderedMounts(mounts))
 
 		// Mount cgroup into the container as readonly, which inherits docker's behavior.
-		s.Mounts = append(s.Mounts, runtimespec.Mount{
+		rspecMount := runtimespec.Mount{
 			Source:      "cgroup",
 			Destination: "/sys/fs/cgroup",
 			Type:        "cgroup",
-			Options:     []string{"nosuid", "noexec", "nodev", "relatime", "ro"},
-		})
+			Options:     []string{"nosuid", "noexec", "nodev", "relatime"},
+		}
+
+		mode := "ro"
+		if cgroupWritable {
+			mode = "rw"
+		}
```

Comment:
If you move this to before line 71, then you only need to change line 75 to `Options: []string{"nosuid", "noexec", "nodev", "relatime", mode},`

Reply:
Thanks @djdongjin! I have made the changes. Can you please review again?